server/broadcast.go

```diff
@@ -116,12 +156,19 @@ func (bsm *BroadcastSessionsManager) completeSession(sess *BroadcastSession) {
 	defer bsm.sessLock.Unlock()
 
 	if existingSess, ok := bsm.sessMap[sess.OrchestratorInfo.Transcoder]; ok {
+		if bsm.lastSess != nil && bsm.lastSess.OrchestratorInfo.Transcoder == sess.OrchestratorInfo.Transcoder {
+			if len(bsm.lastSess.SegsInFlight) > 1 {
+				bsm.lastSess.SegsInFlight = bsm.lastSess.SegsInFlight[1:]
+				// don't place session back into the selector in case a segment is still in flight
```

Comment:
I think we should still set `bsm.lastSess` and update `sessMap` (L170) because the provided `sess` is going to be a new copy (see [this line](https://github.com/livepeer/go-livepeer/blob/897d5760c9aef0519f0dae0b4e34be7888a3a0bc/server/broadcast.go#L442)) that was updated based on an existing session. If we don't make these updates, we could end up using a session for the same orchestrator later on with out-dated metadata fields. We could write a test to verify this by changing the `OrchestratorInfo` message returned with a segment response.